### ISSUE TITLE
No notebook document for `file://*.ipynb` (wsl2)

### ISSUE THREAD

**Issue by thisiswhereitype** [microsoft/vscode#236288]
### Applies To

- [X] Notebooks (.ipynb files)
- [ ] Interactive Window and\/or Cell Scripts (.py files with \#%% markers)

### What happened?

When adding cells: `No notebook document for file://*.ipynb` is prompted. I am using a workspace under WSL2 within the `.vhdx` as a wsl remote.

The notebook goes unresponsive.

### VS Code Version

Version: 1.91.0 (user setup) Commit: ea1445cc7016315d0f5728f8e8b12a45dc0a7286 Date: 2024-07-01T18:52:22.949Z Electron: 29.4.0 ElectronBuildId: 9728852 Chromium: 122.0.6261.156 Node.js: 20.9.0 V8: 12.2.281.27-electron.0 OS: Windows_NT x64 10.0.22631

### Jupyter Extension Version

v2024.6.0

### Jupyter logs

```shell
14:28:18.630 [info] Starting Kernel (Python Path: <repo>/.venv/bin/<interpreter>, Venv, 3.9.19) for '~/git/texconnect/DataTransformationApi/eolt_analysis.ipynb' (disableUI=false)
14:28:19.000 [info] Process Execution: <repo>/.venv/bin/<interpreter> -c "import ipykernel; print(ipykernel.__version__); print("5dc3a68c-e34e-4080-9c3e-2a532b2ccb4d"); print(ipykernel.__file__)"
14:28:19.002 [info] Process Execution: <repo>/.venv/bin/<interpreter> -m ipykernel_launcher --f=/home/~/.local/share/jupyter/runtime/kernel-v2-369SMzJUcplhu2Z.json
    > cwd: //home/~/... # this is unedited after ~
14:28:19.957 [info] Kernel successfully started
```

### Coding Language and Runtime Version

v3.9.19

### Language Extension Version (if applicable)

v2024.8.1

### Running Jupyter locally or remotely?

Local

**Comment by LiekeVanSon** [comment]
I have been experiencing the same issue: the error `No notebook document for file://*.ipynb`  appears after a seemingly random amount of time (between some minutes up to ~1 hour). I first encountered this issue about ~10 days ago. 
I am running jupyter through an ssh tunnel. 

The issue persists:
-  after removing all extensions with `rm -rf ~/.vscode*` , only re-installing the  Jupyter (2024.11.0), and Python (
2024.22.0) extensions. 
 - + disabling _everything_ (i.e. also Pylance) except for the main Jupyter and Python extension.
- updating VS code to the latest Version: 1.96.0
- switching between the pre-release and the release version of Jupyter (2025.1.2024121301). 
- in different notebooks and repo's

**Comment by thisiswhereitype** [comment]
For the time being I can't repro but I often now have slow remote performance related to loading the workspace and tree view actions. (Performance of kernels and scripts is fine).

**Comment by DonJayamanne** [comment]
Would appreciate it if someone can provide concrete steps to repro this issue.
As I'm unable to repro this at my end, I too use notebooks regularly on a daily basis.
Probably something do with some setup/environment issue.
@LiekeVanSon @thisiswhereitype Are you using WSL or remote SSH or the like?

**Comment by thisiswhereitype** [comment]
>  @thisiswhereitype Are you using WSL or remote SSH or the like?

Yes a WSL2 remote on the same host, with the workspace in the VHDX.

Anything like opening expanding directories, loading notebooks and or text files in the editor. The notebook will often get stuck, display cells which don't exist and don't do anything when run, and finally encountering this error. 
For now I just use it to run some Airflow locally so I don't have to interact and develop in a Windows clone of the repo. I recall I did try some profiling but couldn't identify what was wrong.

I don't see much unexpected activity in task manager on the WSL2 process, it seems fairly idle when waiting, I don't see this slowness in a shell or the vscode shell.

@DonJayamanne Would you like a screen capture of what I'm doing/seeing?

My guess at steps for now:
1. Open the WSL remote (warm connect is fine) (I see this in both WSL2 remotes I use).
2. Open the notebook (the bigger the worse)

 @LiekeVanSon  - do you have this characteristic of slowness in WSL2 remotes? Do the above sound like your issue?

**Comment by DonJayamanne** [comment]
> Open the notebook (the bigger the worse)

What do you mean by `worse`.
I thought the issue is you get the message and you cannot open the notebook document. I understand thats bad as is, but could you shed more light on what else gets worse than this.

**Comment by thisiswhereitype** [comment]
I am not sure what I was thinking exactly, probably that it takes longer to load and present the notebook. As I mentioned I have poor performance in VSCode with files anyway - even under VHDX directories.

I gave up and moved to Windows and now only use WSL for less interactive or terminal work.

**Comment by thisiswhereitype** [comment]
I still have this issue on latest stable and WSL and vscode - I think it is slow filesystem related actions from vscode, workspaces take a minute to open, data wrangler stalls trying to open parquets and large notebooks can't run while waiting for formatting etc

**Comment by DonJayamanne** [comment]
If anyone is still running into this please can you share the logs of the errors from console window (use the command `Toggle Developer Tools` to get access to the console window.
